pkg/sql/trace.go

```diff
@@ -43,9 +42,9 @@ type traceNode struct {
 	// around the interaction of SQL with KV. Some of the messages are per-row.
 	kvTracingEnabled bool
 
-	// recording is set if this node started tracing on the session. If it did,
-	// then Close() needs to stop the recording.
-	recording bool
+	// stopTracing is set is set if this node started tracing on the
```

Comment:
is set is set. Also "Close() must close it" is confusing

Why is this a function? It seems that the only purpose is so we can implicitly store `params.p.session` in it (so we're still storing runtime state, we just "hide" it a bit).